src/main/java/com/denizenscript/ddiscordbot/objects/DiscordMessageTag.java

```diff
@@ -471,6 +472,34 @@ public static void register() {
                 mechanism.echoError("Failed to crosspost message: " + ex.getClass().getCanonicalName() + ": " + ex.getMessage());
             }
         });
+        
+        // <--[mechanism]
+        // @object DiscordMessageTag
+        // @name is_pinned
+        // @input ElementTag(Boolean)
+        // @description
+        // Sets whether the message is pinned.
+        // @tags
+        // <DiscordMessageTag.is_pinned>
+        // -->
+        tagProcessor.registerMechanism("is_pinned", false, ElementTag.class, (object, mechanism, input) -> {
+            boolean pinned = input.asBoolean();
+            Message message = object.getMessage();
+            if (!mechanism.requireBoolean()) {
+                return;
+            }

```

Comment:
The check should come _before_ converting the input (should basically always be the first thing)